### ISSUE TITLE
Model training is not reproducible (using google colab on custom dataset)

### ISSUE THREAD

**Issue by suprimnakarmi** [facebookresearch/detectron2#3203]
I am trying to reproduce the same results on different runs (without changing anything on the code), but I could not reproduce the results.  
I have read the documentation and have tried setting the cfg.SEED with a positive value. Also, I have tried applying the code snippet, which I found in a forum, as shown below. However, I could not reproduce the results. 
```
seed = 5

def seed_everything(seed):
    random.seed(seed)
    os.environ['PYTHONHASHSEED'] = str(seed)
    np.random.seed(seed)
    torch.manual_seed(seed)
    torch.cuda.manual_seed(seed)
    torch.backends.cudnn.deterministic = True
    torch.backends.cudnn.benchmark = False
    print("done")
seed_everything(seed)
cfg.SEED=seed
```
I am not sure which other possible random case should be seeded. Please direct me if this issue has been solved somewhere.
I am sharing the code https://colab.research.google.com/drive/1gRbma1hhoiRjG4qK0zJ-N2drpwEQlYac?authuser=4#scrollTo=i57Ytf2qmEsz 

## Environment:

![Screenshot from 2021-06-29 23-20-39](https://user-images.githubusercontent.com/54857456/123843097-3aea7180-d931-11eb-8b75-fc8df86abd9a.png)

**Comment by ppwwyyxx** [comment]
We provide no guarantee that training is deterministic. 

Reducing dataloader NUM_WORKERS and following https://pytorch.org/docs/stable/notes/randomness.html should make training more deterministic

**Comment by suprimnakarmi** [comment]
Thanks, @ppwwyyxx. But I had tried applying NUM_WORKERS to 0 which did not give the expected results. It would be a great help if you could give some hints for where this randomness might have occurred. I have disabled the training augmentation to confirm that randomness is not due to augmentation.
Should I post this to the discussion?

**Comment by Jianghanxiao** [comment]
Hi, I also met this problem. I also try the above stuff, however, I still cannot get the reproducible results. What's the possible reason which cause that detectron2 is not deterministic?

**Comment by Rahul66920** [comment]
Is there any update on this ?
